rudder-core/src/test/scala/com/normation/rudder/services/reports/ComplianceTest.scala

```diff
@@ -0,0 +1,208 @@
+* Copyright 2011 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.rudder.services.reports
+
+import scala.io.Source
+
+import com.normation.rudder.reports.ComplianceMode
+
+import org.junit.runner._
+import org.specs2.mutable._
+import org.specs2.runner._
+
+import net.liftweb.common.Box
+import net.liftweb.common.Failure
+import net.liftweb.common.Full
+import com.normation.rudder.domain.reports.Reports
+import org.joda.time.DateTime
+import com.normation.inventory.domain.NodeId
+import com.normation.rudder.domain.policies.RuleId
+import com.normation.rudder.domain.policies.DirectiveId
+import com.normation.rudder.domain.reports.NodeExpectedReports
+import com.normation.rudder.domain.reports.ExpectedReportsSerialisation
+import com.normation.rudder.domain.reports.NodeConfigId
+import org.joda.time.format.DateTimeFormat
+import org.joda.time.format.DateTimeFormatterBuilder
+import org.joda.time.format.DateTimeParser
+import com.normation.rudder.reports.NodeComplianceMode
+import com.normation.rudder.domain.reports.ComplianceLevel
+
+
+/**
+ *
+ * This file allows to read expected node configuration from a databased export,
+ * the corresponding reports, compute the resulting node status, and compare it
+ * to an expected one.
+ *
+ */
+
+
+@RunWith(classOf[JUnitRunner])
+class ComplianceTest extends Specification {
+
+  val dateParser = ((new DateTimeFormatterBuilder())
```

Comment:
these utility methods may be useful in other tests. Would it make sense to put them in a separate file to be used generically ?